src/main/java/net/neoforged/neoforge/codec/NonNullListCodecs.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.codec;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
+import net.minecraft.core.NonNullList;
+
+public class NonNullListCodecs {
```

Comment:
I would probably add one more extension method here for `ItemStack`s for ease of convenience that only serializes the non-empty ones.